sktime/forecasting/compose/_pipeline.py

```diff
@@ -1437,7 +1449,9 @@ def _predict_interval(self, fh, X, coverage):
                 Upper/lower interval end forecasts are equivalent to
                 quantile forecasts at alpha = 0.5 - c/2, 0.5 + c/2 for c in coverage.
         """
-        X = self._get_forecaster_X_prediction(fh=fh, X=X)
+        if not self.get_tag("ignores-exogeneous-X"):
+            X = self._get_forecaster_X_prediction(fh=fh, X=X)
```

Comment:
not DRY, instead of these repeated lines, I would move it to the start of in `_get_forecaster_X_prediction`

Reply:
Addressed by returning `X` unchanged from `_get_forecaster_X_prediction` in case `ignores-exogeneous-X` is is `True`.

Is that okay, or would you prefer to return `None` instead of unchanged `X`?

Reply:
Realised that this is redundant altogether based on your above comment about `None`. Removing these altogether.